splink/internals/database_api.py

```diff
@@ -385,7 +344,7 @@ def delete_tables_created_by_splink_from_db(self):
             if (
                 key in self._intermediate_table_cache
                 and splink_df.created_by_splink
-                and key == splink_df.physical_name
+                and key == splink_df.templated_name
```

Comment:
<img width="829" height="179" alt="image" src="https://github.com/user-attachments/assets/2ff80adf-ded7-4d72-95f5-17e1355a81c9" />

Previously we were guaranteed to have the physical name in the cache because all tables were cached (see above, which is what the cache used to look like during the [table deletions test](https://github.com/moj-analytical-services/splink/blob/50dc237cbdb53efad7379e013ccd042f13f718f0/tests/test_caching_tables.py#L226))

Now it's much more possible that there are tables created by Splink that are not in the cache, so we need to be more careful to run `drop_table_from_database_and_remove_from_cache`